### ISSUE TITLE
0.49 RC: AttributeError: module 'numba' has no attribute 'numpy_support'

### ISSUE THREAD

**Issue by hameerabbasi** [numba/numba#5466]
The commit https://github.com/pydata/sparse/commit/b642e115d5a66db209abf812637fad752c13ed19 of PyData/Sparse passes in Numba 0.48.

In 0.49, it fails with the message in the title.

**Comment by hameerabbasi** [comment]
Further digging reveals that `numpy_support` is available, but only once it's imported:

```python
>>> import numba
>>> numba.numpy_support
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'numba' has no attribute 'numpy_support'
>>> import numba.numpy_support
<stdin>:1: NumbaDeprecationWarning: An import was requested from a module that has moved location.
Import requested from: 'numba.numpy_support', please update to use 'numba.np.numpy_support' or pin to Numba version 0.48.0. This alias will not be present in Numba version 0.50.0.
>>> numba.numpy_support
<module 'numba.numpy_support' from '/Users/hameerabbasi/miniforge3/envs/sparse/lib/python3.8/site-packages/numba/numpy_support.py'>
```

**Comment by stuartarchibald** [comment]
Thanks for the report. I think its existence may well have been a side effect of import chains.

Prior to 0.49: `numba.__init__` imported `numba.special` imported `numba.typing` imported `numba.typing.context` imported `numba.typing.typeof` imported `numba.numpy_support`.

The 0.48 release doesn't have it in `__init__` https://github.com/numba/numba/blob/release0.48/numba/__init__.py

thoughts ?

**Comment by hameerabbasi** [comment]
Well, there's code out there I'd like to not break on day 0, so it'd be nice if it was added to `__init__.py` for 0.49 and dropped in 0.50 if necessary.

**Comment by stuartarchibald** [comment]
If I can find a reasonable way to ensure that a warning notice about the deprecation is emitted, and only for the case above, then I think that this can be patched in for 0.49.
